src/components/forms/types/FieldBinder.ts

```diff
@@ -0,0 +1,3 @@
+import { Consumer } from "../../../utils/consumer";
+import { Field } from "./Field";
+export type FieldBinder<S, T> = (state: S, renderApp: Consumer<S>) => Field<T>;
```

Comment:
a field binder is something that will _eventually_ be a field, but it needs to be _bound_ to the `state` and `renderApp` so that updating the field will have the effect of updating the state and rerendering...

see a use case of "bindField" or "bindFields" to see a fieldBinder being automatically turned into a field, where it can in turn be consumed by a form control (like an input text, for example)